README.md

```diff
@@ -74,30 +90,105 @@ If you work with [Q-cli](https://github.com/nzzdev/Q-cli) there is a config file
 
 Of course with another mapbox configuration you have to adjust the LAYER_CONFIGS defintion as well so that it fits your needs.
 
-## Installation
-
-```bash
-$ npm install
-$ npm run build
-```
+[to the top](#table-of-contents)
 
 ## Development
 
-Install the [Q cli](https://github.com/nzzdev/Q-cli) and start the Q dev server:
+Start the Q dev server:
 
 ```
-$ Q server
+npx @nzz/q-cli server -c ./dev-config/server-config.js
 ```
 
 Run the Q tool:
+
 ```
-$ npm run dev
+node dev.js
 ```
 
-## Implementation details
+[to the top](#table-of-contents)
+
+## Testing
+
+The testing framework used in this repository is [Code](https://github.com/hapijs/code).
+
+Run the tests:
+
+```
+npm run test
+```
+
+### Implementing a new test
+
+When changing or implementing...
+
+- A `route`, it needs to be tested in the `e2e-tests.js` file
+- Something on the frontend, it needs to be tested in the `dom-tests.js` file
+
+[to the top](#table-of-contents)
+
+## Deployment
+
+We provide automatically built docker images at https://hub.docker.com/r/nzzonline/q-map/.
+There are three options for deployment:
+
+- use the provided images
+- build your own docker images
+- deploy the service using another technology
+
+### Use the provided docker images
+
+1. Deploy `nzzonline/q-map` to a docker environment
+2. Set the ENV variables as described in the [configuration section](#configuration)
+
+[to the top](#table-of-contents)
+
+## Functionality
+
 The tool structure follows the general structure of each Q tool. Further information can be found in [Q server documentation - Developing tools](https://nzzdev.github.io/Q-server/developing-tools.html).
 
+Q Map uses the [svelte framework](https://svelte.technology/guide) to render the markup on server-side. Additionally a jspm bundle is transpiled for all interactive parts on client-side.
+
+### Options
+
+#### baselayer
+
+This option allows to switch between six different basemaps.
+
+#### initialZoomLevel
+
+Allows to specify the zoomlevel of the basemap.
+
+#### minimap and minimapInitialZoomOffset
+
+Show a minimap and specify a minimap zoomlevel.
+
+#### labelsBelowMap
+
+Allows to show the marker labels below the map.
+
+#### showLegend
+
+This options allows to show the legend. The legend only shows something if features where added as geojson.
+
+#### Display Options
```

Comment:
rename `displayOptions`

Reply:
We decided to leave this, because its not a property but a title